proxy/healthy_endpoints_test.go

```diff
@@ -0,0 +1,104 @@
+package proxy
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/loadbalancer"
+	"github.com/zalando/skipper/routing"
+)
+
+func initiliaze(hosts []string) (*routing.EndpointRegistry, *routing.Route) {
+	endpointRegistry := routing.NewEndpointRegistry(routing.RegistryOptions{})
+	eskipRoute := eskip.Route{BackendType: eskip.LBBackend, LBAlgorithm: loadbalancer.PowerOfRandomNChoices.String()}
+	for i := range hosts {
+		eskipRoute.LBEndpoints = append(eskipRoute.LBEndpoints, fmt.Sprintf("http://%s", hosts[i]))
+	}
+
+	route := &routing.Route{
+		Route:       eskipRoute,
+		LBEndpoints: []routing.LBEndpoint{},
+	}
+	rt := loadbalancer.NewAlgorithmProvider().Do([]*routing.Route{route})
+	route = rt[0]
+	endpointRegistry.Do([]*routing.Route{route})
+
+	return endpointRegistry, route
+}
+
+func TestFilterHealthyEndpointsReturnsAllEndpointsWhenAllAreHealthy(t *testing.T) {
+	hosts := []string{"host1:80", "host2:80", "host3:80"}
```

Comment:
You may use .test domain here e.g. host1.test, see https://www.rfc-editor.org/rfc/rfc2606.html#section-2